File: 1475.py
from typing import List
from collections import deque

class Solution:
    def finalPrices(self, prices: List[int]) -> List[int]:
        # Create a copy of prices array to store discounted prices
        result = prices.copy()

        stack = deque()

        for i in range(len(prices)):
            # Process items that can be discounted by current price
            while stack and prices[stack[-1]] >= prices[i]:
                # Apply discount to previous item using current price
                result[stack.pop()] -= prices[i]
            # Add current index to stack
            stack.append(i)

        return result


if __name__ == '__main__':
    test_cases = ( [[8,4,6,2,3], [4,2,4,2,3]], [[1,2,3,4,5], [1,2,3,4,5]])
    for case in test_cases:
        print('Input: ' + str(case[0]) + '\nExpected output: ' + str(case[1]) + '\nActual output: ' + str(Solution().finalPrices(case[0])) + '\n')

File: 1475.js
/**
 * @param {number[]} prices
 * @return {number[]}
 */
var finalPrices = function(prices) {
    let returnArray = [];
    let temp = [];
    for(let i = 0; i < prices.length; i++)
    {
        
    }
};

testCases = [ [[8,4,6,2,3], [4,2,4,2,3]], [[1,2,3,4,5], [1,2,3,4,5]], [[10,1,1,6], [9,0,1,6]]];
for(const test of testCases)
{
    console.log('Input: ' + test[0] + '\nExpected output: ' + test[1] + '\nActual output: ' + finalPrices(test[0]) + '\n');
}